src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,96 @@
+import { useTodos } from '../../utils/TodoContext';
+import { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/types';
+
+interface Props {
+  key: number;
+  todo: Todo;
+}
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { setTodos, removeTodo } = useTodos();
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+  const toggleCompletedTodo = (id: number) => {
+    setTodos(prevTodos =>
+      prevTodos.map(prev =>
+        prev.id === id ? { ...prev, completed: !prev.completed } : prev,
+      ),
+    );
+  };
+
+  const toggleEditTodo = (id: number) => {
+    setEditingTodoId(id);
+  };
+
+  const handleEditInputChange = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    id: number,
+  ) => {
+    const newText = e.target.value;
+
+    setTodos(prevTodos =>
+      prevTodos.map(prev =>
+        prev.id === id ? { ...prev, title: newText } : prev,
+      ),
+    );
+  };
+
+  const handleEditInputKeyUp = (
+    e: React.KeyboardEvent<HTMLInputElement>,
+    id: number,
+  ) => {
+    if (e.key === 'Enter') {
+      const newText = e.currentTarget.value.trim();
+
+      setTodos(prevTodos =>
+        prevTodos.map(prev =>
+          prev.id === id ? { ...prev, title: newText } : prev,
+        ),
+      );
+
+      if (newText === '') {
```

Comment:
```suggestion
      if (!newText) {
```
You don't need to compare it with an empty string
https://developer.mozilla.org/en-US/docs/Glossary/Falsy